.github/workflows/scenario-validator.yml

```diff
@@ -0,0 +1,74 @@
+# Runs scenario tests for a given scenario
+name: Scenario validator
+on:
+  workflow_dispatch:
+    inputs:
+      scenario_name:
+        description: 'Name of the scenario to test'
+        required: true
+
+jobs:
+  get-go-version:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./test/acceptance
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+    - name: Determine Go version
+      id: get-go-version
+      run: |
+        echo "Building with Go $(cat .go-version)"
+        echo "go-version=$(cat .go-version)" >> "$GITHUB_OUTPUT"
```

Comment:
Should we extract this out in different file named `reusable-get-go-version.yaml` because this is also used in `.github/workflows/nightly-ecs-examples-validator.yml` ?

Reply:
I think we do this. Please check - https://github.com/hashicorp/consul-enterprise/blob/76b5de5039e9101416d5722046784e29d741d57f/.github/workflows/reusable-get-go-version.yml

Reply:
If its being done, then we should do it for other jobs as well like `terraform-fmt` and `go-fmt`. I have addressed this. Thanks for bringing this up.